File: client/test/tests/router/pages.js
export const BASE = "/test/test.html";
const START = "./pages".length;

export const pages = (() => {
  return Object.fromEntries(
    Object.entries(import.meta.glob("./pages/**/*.js")).map(([k, load]) => {
      const path = `${BASE}${k.slice(START, -3)}`;
      return [path, load];
    })
  );
})();

const home = pages[`${BASE}/home`]
if (home) {
  pages[`${BASE}`] = home
}

File: _history/server/server_code/20260504t2200/router/_pages.py
from ..tools import access, meta

# Default page
START = "start"

# Declared pages (page:spec items)
pages = dict(
    # Form routes
    iworker=dict(require=lambda: meta.PROD or access()),
    login=dict(),
    plot=dict(),
    signup=dict(),
    start=dict(),
    test=dict(require=lambda: meta.DEV or (meta.PROD and access())),
    # View routes
    blog=dict(view="blog/blog.jinja"),
    foo=dict(view="foo/foo.jinja"),
)

# 'page' -> 'page'
redirects = {
    "user-login": "login",
    "user-signup": "signup",
}
# Browser routes
redirects.update(about=START, front=START, home=START, stats=START)


# 'path' -> 'path'
rewrites = {
    "/front": "/front/index.html",
}
